test/tests.ts

```diff
@@ -1111,70 +1157,76 @@ describe("passport-saml /", function () {
       },
     ];
 
-    var server;
+    let server: Server;
 
-    function testForCheck(check) {
-      return function (done) {
-        var app = express();
+    function testForCheck(check: SAMLCheck) {
+      return function (done: Mocha.Done) {
+        const app = express();
         app.use(bodyParser.urlencoded({ extended: false }));
         app.use(passport.initialize());
-        var config = check.config;
+        const config = check.config;
         config.callbackUrl = "http://localhost:3033/login";
         config.entryPoint = "https://wwwexampleIdp.com/saml";
-        var profile = null;
+        let profile = null;
         passport.use(
-          new SamlStrategy(config, function (_profile, done) {
+          new SamlStrategy(config, function (_profile: Profile, done: VerifiedCallback) {
             profile = _profile;
             done(null, profile);
-          })
+          } as VerifyWithoutRequest)
         );
 
-        var userSerialized = false;
+        let userSerialized = false;
         passport.serializeUser(function (user, done) {
           userSerialized = true;
           done(null, user);
         });
 
         app.get(
           "/login",
+          // @ts-expect-error the problem of passport typings
```

Comment:
Is this something I can help fix over at DefinitelyTyped?

Reply:
I think strategy typings need major rewrite, I don't think they need impede our tests

Reply:
I fixed this and will try to push something up later today.

Reply:
I fixed this by using the types that we have defined in our project.